### ISSUE TITLE
add namespace sgd, and sub-namespace things

### ISSUE THREAD

**Issue by dustinvtran** [airoldilab/sgd#72]
I like this approach to avoid names clashing during implementation. For instance (arbitrarily chosen), see Stan's approach as in the file https://github.com/stan-dev/stan/blob/develop/src/stan/mcmc/sample.hpp.

**Comment by ptoulis** [comment]
you mean the namespaces?

**Comment by dustinvtran** [comment]
Yup, in the same way they have `stan::mcmc::sample`, we should have something such as `sgd::glm_experiment`
